sys/riotboot/flashwrite.c

```diff
@@ -178,10 +173,13 @@ int riotboot_flashwrite_invalidate(int slot)
         return -2;
     }
 
-    uint8_t data_flash[4];
-    memset(data_flash, INVALIDATE_HDR, sizeof(data_flash));
+    /* invalidate header (checksum and magic number must be invalidated),
+       write the whole header to avoid running in memory alignment issues
+       with FLASHPAGE_WRITE_BLOCK_SIZE */
+    riotboot_hdr_t tmp_hdr = {0};
```

Comment:
shouldn't that be filled with `FLASHPAGE_ERASE_STATE`

Reply:
true, all that to not use the variable